logviewer/src/main/java/org/togetherjava/logwatcher/users/UserRepositoryImpl.java

```diff
@@ -0,0 +1,108 @@
+package org.togetherjava.logwatcher.users;
+
+import org.springframework.stereotype.Component;
+import org.togetherjava.logwatcher.accesscontrol.Role;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.Userroles;
+import org.togetherjava.tjbot.db.generated.tables.pojos.Users;
+import org.togetherjava.tjbot.db.generated.tables.records.UserrolesRecord;
+import org.togetherjava.tjbot.db.generated.tables.records.UsersRecord;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.togetherjava.tjbot.db.generated.tables.Users.USERS;
+
+@Component
+@SuppressWarnings("java:S1602") // Curly Braces are necessary here
```

Comment:
Why, where? Is it really necessary everywhere in the class? It doesnt seem like to me.

Please check if you can just move the suppression to the actual statements instead of the whole class.

Reply:
I mean I could also write it 9 times (each time I talk to the DB), but then I would just rather set the Ignore on the whole class

Reply:
Then lets talk about those 8 cases. Why is it needed there? I do not see it. They are single statements, a curly is obsolete.

Reply:
Because of the Design of the Database-class it's an ambiguous match:
[Invalid](https://imgur.com/EPAT4s1)

Reply:
Interesting. Cant you add some cast to it to make it clear?

(lemme loop in @I-Al-Istannen for this, who designed the DB class)

Reply:
Yes cast's are possible, but readability suffers greatly from them imo, see:
[Casts](https://imgur.com/5fGVr8C.jpg)

Reply:
Without the braces java can not correctly infer that you actually want to return nothing there and it picks up the function variant of the method as another possible match. Sonarlint isn't smart enough to figure out that the braces are actually needed here and complains anyways.

I probably said this earlier but the only real solution (besides fixing Sonar) would be to rename one of the methods in the database class, which should probably be alright to do. We can do that after the current PRs are merged.

Reply:
> that you actually want to return nothing there

But I do return something

Reply:
That's the same problem but in reverse, the statement still applies. If you are interested you can look at JLS §15.27.3 I think. A Method invocation is an `Expression Statement` and its value might be discarded. Following the rules in the mentioned JSL paragraph, the invocation could either be an expression that returns its value (matching the Function method) or an expression statement that discards its value (matching the consumer method).

Java can not tell the difference and so you are forced to disambiguate. Adding the block solves the issue as you then have a void-compatible block that is not value-compatible and therefore the compiler can figure out what you wanted. That Sonar doesn't perform the same analysis is a bug.

My previous comment still applies in its entirety.

Reply:
Can this be marked as resolved? The Curly Braces and the Suppression are necessary here.

Reply:
Yea, let's fix this in a future PR.